content/blog/pragmatism-over-perfection.md

```diff
@@ -0,0 +1,120 @@
+---
+title: "Pragmatism Over Perfection"
+authorId: "aleric"
+date: 2025-04-21
+draft: false
+featured: true
+weight: 1
+---
+
+We as Engineers often chase perfection.
+
+It fuels our curiosity, sharpens our skills, and makes us feel good about the things we build. But at the same time, it can be a double-edged sword, sometimes slowing us down or distracting us from what actually matters: **delivering business value**.
+
+That’s where pragmatism comes in.
+
+This is a story about me getting that lesson reinforced with a simple task I was working on: KYC Image Tagging.
+
+## The Problem
+
+I had worked with the ML team to develop two models for tagging KYC signature images as **valid** or **invalid** — one based on a CNN, the other on a Decision Tree.
+
+Once trained, we ran both models on a dataset of around **1.2 lakh images**. They disagreed on about **17%** of them — and the only way to figure out which model was better was to manually tag those images and compare.
+
+The ops team was ready to help. All we needed now was a simple interface for them to actually do the tagging.
+
+So I got to work exploring image tagging tools.
+
+## Exploring the Right Tools for the Job
+
+We had a couple of requirements for this process:
+
+1. **Security and Access Control:** The signature images were PII, so we needed secure login, proper access control, and audit logs.
+
+2. **Speed over Longevity:** This was going to be mostly a one time activity, and there was some pressure to get this done as soon as possible.
+
+I quickly found a couple of open-source tagging tools available and started evaluating each one so that I could present viable options to the stakeholders. For each, I looked into pros, cons and how to get them up and running securely and so that only the Ops team could access them.
+
+However there were some constraints:
+
+1. **Missing Security Features:** Most open-source tools lacked the security and access controls we needed, at least in their free versions. And buying licenses didn’t make sense for a one-off task, this meant that I would need to somehow implement security and access control.
+
+2. **Time pressure:** The images were ready to be tagged. The stakeholders were waiting. And the Ops team was blocked on me to provide them with an interface. Setting up the infrastructure, securing it, and onboarding users would take time we did not have.
+
+These constraints made the "right" tools impractical for our needs.
+
+## A Hack that Worked
+
+This is when the shift happened. I stepped back and looked at what we already had, and an idea struck.
+
+We were already using Databricks internally. What if it could be turned into a tagging tool?
```

Comment:
Could you elaborate on the usage of databricks ? i.e. it stored what sort of data that was relevant to this task?